packages/renderer/src/lib/objects/KubernetesObjectsList.svelte

```diff
@@ -0,0 +1,159 @@
+<script lang="ts">
+import { faTrash } from '@fortawesome/free-solid-svg-icons';
+import type { KubernetesObject } from '@kubernetes/client-node';
+import type { TableColumn, TableRow } from '@podman-desktop/ui-svelte';
+import { Button, FilteredEmptyScreen, NavPage, Table } from '@podman-desktop/ui-svelte';
+import { onDestroy, onMount, type Snippet } from 'svelte';
+import { type Readable, type Unsubscriber, type Writable } from 'svelte/store';
+
+import type { IDisposable } from '../../../../main/src/plugin/types/disposable';
+import { withBulkConfirmation } from '../actions/BulkActions';
+import KubeActions from '../kube/KubeActions.svelte';
+import { listenResources } from '../kube/resources-listen';
+import KubernetesCurrentContextConnectionBadge from '../ui/KubernetesCurrentContextConnectionBadge.svelte';
+
+export interface Kind {
+  resource: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  transformer: (o: KubernetesObject) => any;
+  delete: (name: string) => Promise<void>;
+  isResource: (o: KubernetesObject) => boolean;
+  searchPatternStore: Writable<string>;
```

Comment:
For the current config maps & secrets we only used two search pattern stores because of the underlying implementation - there's really only one filter bar / one search term that we apply to both. Could this be renamed legacySearchPatternStore to be clear the intent is that it would be removed later?

IMHO having the search term / pattern store alongside each object store was one of the things we got 'wrong' in the current implementation. We don't have a shared list of filtered deployments, we have multiple pages that each supporting filtering. i.e. filtering should be managed by KubernetesObjectsList here, not the stores or shared between lists.

Reply:
ok, renaming to `legacySearchPatternStore`, and I agree that we can find a better pattern for the search